apps/ehr/src/features/external-labs/components/LabsAutocomplete.tsx

```
@@ -0,0 +1,118 @@
+
+  useEffect(() => {
+    async function getLabs(oystehr: Oystehr): Promise<void> {
+      if (!oystehr) {
+        return;
+      }
+
+      setLoading(true);
+
+      const labOrgsGuids: string[] = [];
+
+      try {
+        const organizationSearch = (
+          await oystehr.fhir.search<Organization>({
+            resourceType: 'Organization',
+            params: [{ name: 'type', value: `${LAB_ORG_TYPE_CODING.system}|${LAB_ORG_TYPE_CODING.code}` }],
+          })
+        ).unbundle();
+        organizationSearch.forEach((org) => {
+          const labGuid = org.identifier?.find((id) => id.system === OYSTEHR_LAB_GUID_SYSTEM)?.value;
+          if (labGuid) labOrgsGuids.push(labGuid);
+        });
+      } catch (e) {
+        console.error('error getting lab organizations', e);
+      }
+
+      if (labOrgsGuids) {
+        try {
+          const token = await getAccessTokenSilently();
+          const projectId = import.meta.env.VITE_APP_PROJECT_ID;
+          const labIds = labOrgsGuids.join(',');
+          let cursor = '';
+          const items: OrderableItemSearchResult[] = [];
+
+          do {
+            const url = `${OYSTEHR_LAB_ORDERABLE_ITEM_SEARCH_API}?labIds=${labIds}&limit=100&cursor=${cursor}`;
+            const orderableItemsSearch = await fetch(url, {
+              method: 'GET',
+              headers: {
+                'x-zapehr-project-id': projectId,
+                Authorization: `Bearer ${token}`,
+              },
+            });
+            const response = await orderableItemsSearch.json();
+            const orderableItemRes = response.orderableItems as OrderableItemSearchResult[];
+            items.push(...orderableItemRes);
+            cursor = response?.metadata?.nextCursor || '';
+          } while (cursor);
+
+          setLabs(items);
+        } catch (e) {
+          console.error('error fetching orderable items', e);
+        }
+      }
+
+      setLoading(false);
+    }
+
+    if (oystehr) {
```

Can we add a todo here for error handling? looks like we'll just silently fail here and throw some stuff in the console if any step in this loading process fails.